project/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+from ordered_model.models import OrderedModel
+
+from neatplus.models import TimeStampedModel, UserStampedModel
+
+
+class Project(TimeStampedModel, UserStampedModel, OrderedModel):
+    class VisibilityChoice(models.TextChoices):
+        PUBLIC = "public"
+        PUBLIC_WIITHIN_ORGANIZATION = "public_within_organization"
+        PRIVATE = "private"
+
+    class StatusChoice(models.TextChoices):
+        PENDING = "pending"
+        ACCEPTED = "accepted"
+        REJECTED = "rejected"
+
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    organization = models.ForeignKey(
+        "organization.Organization", on_delete=models.CASCADE, related_name="projects"
+    )
+    visibility = models.CharField(max_length=26, choices=VisibilityChoice.choices)
+    users = models.ManyToManyField(
+        settings.AUTH_USER_MODEL,
+        related_name="projects",
+        through="ProjectUser",
+        through_fields=("project", "user"),
+    )
+    status = models.CharField(
+        max_length=8,
+        choices=StatusChoice.choices,
+        default=StatusChoice.PENDING,
+        editable=False,
+    )
+    context = models.ForeignKey(
+        "context.Context", on_delete=models.CASCADE, related_name="projects"
+    )
+
+    class Meta(OrderedModel.Meta):
+        pass
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
```

Comment:
should we create a mixin for created_fields generation? Might come in handy as we need such feature often.